model/inline/src/main/scala/aqua/model/inline/TagInliner.scala

```diff
@@ -79,33 +79,49 @@ object TagInliner extends Logging {
       prefix: Option[OpModel.Tree] = None
     ) extends TagInlined[S](prefix)
 
+    /**
+     * Tag inlining emitted computation
+     * that envelopes children computation
+     *
+     * @param model computation producing model based on children computation
+     */
+    case Around[S](
+      model: State[S, Chain[OpModel.Tree]] => State[S, OpModel.Tree],
+      aroundChildren: State[S, OpModel.Tree] => State[S, OpModel.Tree],
```

Comment:
Is there really a need to separate this into two functions?
Maybe lets make `model: Chain[State[S, OpModel.Tree]]` and inside this function can do all the same.

btw, it is a cool source of optimization, e.g. `if` can skip the whole branch computation if it can prove that condition is always true or false!